source/agora/consensus/PreImage.d

```diff
@@ -343,15 +343,18 @@ public struct PreImageCycle
         is part of consensus.
 
         Params:
-          secret = The secret key of the node, used as part of the hash
-                   to generate the cycle seeds
+            secret = The secret key of the node, used as part of the hash
+                to generate the cycle seeds
+            calc_index = The flag to determine to calculating the cycle index
+                or just getting a random seed which is the first pre-image
 
         Returns:
           The hash of the current enrollment round
 
     ***************************************************************************/
 
-    public Hash consume (scope const ref Scalar secret) @safe nothrow
+    public Hash consume (scope const ref Scalar secret, bool calc_index = true)
+        @safe nothrow
```

Comment:
Since the name is `consume`, the `calc_index` argument is at odds with the goal of the function.
I would rename it to `populate` and rename `calc_index` to `consume`.
The description also needs to be updated.